[DOCS] Endpoint self-healing rollback (#2267)
elastic/security-docs: docs/getting-started/self-healing-rollback.asciidoc

[[self-healing-rollback]]
= Configure self-healing rollback for Windows endpoints

> {endpoint-cloud-sec}'s self-healing feature rolls back file changes and processes on Windows endpoints when a prevention alert is generated by enabled protection features. All activity on the host reverts to its state five minutes before the prevention alert.

Reviewer: Process remediation is planned but not implemented yet. Also, afaik there is nuance with the 5 min value. For example, modified files are rollbacked to most recent snapshot, which can be much longer than 5 minutes

Author: Thanks @joe-desimone for the extra info! Is this a standard Elasticsearch snapshot, or something unique to Endpoint? And can we say anything more about how often the snapshots are taken, what triggers them, etc.? To give readers a sense of what to expect.

Reviewer: These snapshots are unique to endpoint. We use built in volume shadow copy functionality to windows https://docs.microsoft.com/en-us/windows-server/storage/file-server/volume-shadow-copy-service. I believe we take a snapshot every 2 hours. We might tweak this slightly or even make it user configurable in the future though

Reviewer: Cc @bit-envoy

Author: @joe-desimone @bit-envoy Wondering if this covers it -- I don't think we need too much detail about Endpoint's inner workings, but a general time window might help users know how far back to expect rollback, for example to confirm any incidental changes:
```suggestion
{endpoint-cloud-sec}'s self-healing feature rolls back file changes on Windows endpoints when a prevention alert is generated by enabled protection features. File changes on the host revert to their state up to two hours before the prevention alert.
```

Reviewer: I think this is accurate but it would be best if @bit-envoy can confirm

Reviewer 2: It is reverting all changes that happened in five minutes to disk state that was captured during taking snapshot (which might max 2h)

Author: Thanks, @bit-envoy! Is it accurate to say that it rolls back five minutes _from when the alert was created_? Or is there another reference point for the five minutes?

> {endpoint-cloud-sec}'s self-healing feature rolls back file changes and processes on Windows endpoints when a prevention alert is generated by enabled protection features. All activity on the host reverts to its state five minutes before the prevention alert.

Author: I think "Endpoint and Cloud Security feature" is appropriate here, because it's part of that integration and this sentence is talking about what _the feature_ is and does, not how it's configured. After the rename of the integration, there isn't really anything called "Endpoint Security" anymore. @ferullo @joe-desimone thoughts?

> {endpoint-cloud-sec}'s self-healing feature rolls back file changes and processes on Windows endpoints when a prevention alert is generated by enabled protection features. All activity on the host reverts to its state five minutes before the prevention alert.

Reviewer 3: I think Endpoint and Cloud Security is appropriate here.

> This can help contain the impact of malicious activity, as {endpoint-cloud-sec} not only stops the activity but also erases any attack artifacts deployed prior to detection.

Reviewer 4: Same question about terminology. Isn't Elastic Endpoint doing the stopping of malicious activity? 🤔

Author: "Elastic Endpoint" seems to me more like an internal piece of technology than something a customer directly experiences. I was using the integration name here to more generally say that _the product_ or _the integration_ stops and rolls back the attack. But @ferullo @joe-desimone etc please let us know what you think.

Reviewer 3: the integration name seems correct here.

Self-healing rollback is a https://www.elastic.co/pricing[Platinum or Enterprise subscription] feature, and it's only supported for Windows endpoints.

[CAUTION]
====

> This feature can cause data loss, since it reverts _all_ recent changes on the host, not just changes directly related to a threat.

Reviewer: In its current implementation yes permanent data loss can occur because we delete recently added files or overwrite changes with an old copy. 

It is not accurate here that _all_ recent changes are reverted. For example, we have a set of filters which limit targeted files to paths which are most likely related to the threat. However, it is accurate to say that there can be incidental actions that are unrelated to the threat.

> This feature can cause data loss, since it reverts _all_ recent changes on the host, not just changes directly related to a threat.

Author: @joe-desimone @nastasha-solomon Let me know if this conveys those nuances:
```suggestion
This feature can cause permanent data loss, since it overwrites recent changes and deletes recently added files on the host. Self-healing rollback targets the changes related to a detected threat, but it may also include incidental actions that aren't directly related to the threat.
```

> This feature can cause permanent data loss, since it overwrites recent changes and deletes recently added files on the host. Self-healing rollback targets the changes related to a detected threat, but it may also include incidental actions that aren't directly related to the threat.

Reviewer 4: ```suggestion
This feature can cause permanent data loss since it overwrites recent changes and deletes recently added files on the host. Self-healing rollback targets the changes related to a detected threat, but may also include incidental actions that aren't directly related to the threat.
```
